File: sources/opencv-samples/14_HistogramEqualizationGray/main.py
import cv2 as cv
import sys
import argparse
from time import time


def getTime():
	return time()


def getTimeDelta(t):
	t = 1000 * (getTime() - t)
	return round(1000 * t) / 1000


def main():
	args = parser.parse_args()
	
	path = cv.samples.findFile(args.img_path)
	
	if not path:
		sys.exit(f"Image {args.img_path} not found")
	
	image = cv.imread(path)

	if image is None:
		sys.exit(f"Could not read the image {path}")

	window1 = "Image gray"
	window2 = "Equalized"

	# Convertir en niveaux de gris
	gray = cv.cvtColor(image, cv.COLOR_BGR2GRAY)

	t = getTime()

	# Appliquer l'égalisation d'histogramme
	equalized = cv.equalizeHist(gray)

	print(f"duration: {getTimeDelta(t)} ms")

	cv.imshow(window1, gray)
	cv.imshow(window2, equalized)

	cv.waitKey(0)
	cv.destroyAllWindows()


if __name__ == '__main__':
    parser = argparse.ArgumentParser(description="Histogram Equalization Gray main.py CLI")
    parser.add_argument("img_path", type=str, help="image path to open")
    main()

File: sources/opencv-samples/14_HistogramEqualizationGray/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;


double getTime()
{
    return (double)getTickCount();
}

double getTimeDiff(double time_ori)
{
    return 1000 * (getTime() - time_ori) / getTickFrequency();
}

int main(int argc, char** argv) 
{
    if (argc != 2)
    {
        printf("Usage: ./main.out <image_path>");
        return -1;
    }

    Mat image = imread(argv[1]);

    if (!image.data)
    {
        cout << "Could not read the image " << argv[1] << endl;
        return -1;
    }

    string window1 = "Image gray";
    string window2 = "Equalized";

    double t = getTime();

    // Convertir en niveaux de gris
    cv::Mat gray;
    cv::cvtColor(image, gray, cv::COLOR_BGR2GRAY);
    
    // Appliquer l'égalisation d'histogramme
    cv::Mat equalized;
    cv::equalizeHist(gray, equalized);

    cout << "duration : " << getTimeDiff(t) << " ms" << endl;

    imshow(window1, image);
    imshow(window2, equalized);

    waitKey(0);
    destroyAllWindows();
    return 0;
}
